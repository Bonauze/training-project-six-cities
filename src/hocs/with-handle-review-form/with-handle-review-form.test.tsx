import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';

import withHandleReviewForm from './with-handle-review-form';
import testStore from '../../test-mocks/store';
import configureStore from 'redux-mock-store';

interface MockComponentProps {
  className: string;
}

const MockComponent: React.FunctionComponent<MockComponentProps> = (props: MockComponentProps) => {
  const {className} = props;

  return (
    <div className={className} />
  );
};

const MockComponentWrapped = withHandleReviewForm(MockComponent);

const mockStore = configureStore([]);

it(`withHandleReviewForm renders correctly`, () => {
  const testMockStore = mockStore(testStore);

  const tree = renderer.create(
      <Provider store={testMockStore}>
        <MockComponentWrapped
          className={`test-class`}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
