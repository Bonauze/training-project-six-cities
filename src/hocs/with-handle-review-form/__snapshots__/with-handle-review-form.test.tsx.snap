// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withHandleReviewForm renders correctly 1`] = `
<div
  className="test-class"
/>
`;
